### PR TITLE
Add accessible text content

### PR DESCRIPTION
**Pull request by krokodok**
For #76, inspired by https://github.com/wearerequired/swisscom-magazine/issues/1256.

---

https://github.com/wearerequired/rest-likes/compare/master...feature/accessible-text-content#diff-3aa09978cb18aff029d3dd6535a2b9ee0a8c4df561ceae68517deb6bac6ffa7aR182-R195

https://github.com/wearerequired/rest-likes/compare/master...feature/accessible-text-content#diff-3aa09978cb18aff029d3dd6535a2b9ee0a8c4df561ceae68517deb6bac6ffa7aR329-R342

I wanted to include these strings as filters, the same as here: https://github.com/wearerequired/rest-likes/blob/feature/accessible-text-content/inc/Plugin.php#L155-L167

But because I don't know the `count` yet in the Php definition, I have to translate the string during runtime.

### CODE REVIEW

File: inc/Plugin.php [hunk 1]
```diff
@@ -165,6 +165,10 @@ protected function get_object_types_script_data() {
 					 */
 					'unlike' => apply_filters( 'rest_likes.button_text.unlike', _x( 'Unlike', 'verb', 'rest-likes' ), $object_type ),
 				],
+				'html'       => [
+					'visual_text'        => '<span aria-hidden="true">%s</span>',
+					'screen_reader_text' => '<span class="screen-reader-text">%s</span>',
+				],
```
Comment (ocean90) [inline inc/Plugin.php:171]: Let's remove this since we can't provide the actual strings here.

File: js/src/index.js [hunk 2]
```diff
@@ -309,7 +326,24 @@ $document.on( 'heartbeat-tick', ( event, data ) => {
...
+				likeButtonScreenReaderText = sprintf(
+					/* translators: $s = number of likes */
+					_n( '%s like', '%s likes', count, 'rest-likes' ),
+					countFormatted
+				);
+			}
+
+			likeButtonCount.innerHTML =
+				sprintf( objectTypeData.html.visual_text, countFormatted ) + // eslint-disable-line @wordpress/valid-sprintf
+				sprintf( objectTypeData.html.screen_reader_text, likeButtonScreenReaderText ); // eslint-disable-line @wordpress/valid-sprintf
```
Comment (ocean90) [inline js/src/index.js:346]: Use `document.createElement()` with `.textContent` for the text.

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: inc/Controller.php [hunk 3]
```diff
@@ -520,13 +520,28 @@ public function get_like_count( $object_id ) {
 	public function get_like_count_html( $object_id ) {
 		$likes = $this->get_like_count( $object_id );
 
+		if ( 1 === $likes ) {
+			$likes_text = sprintf(
+				/* translators: $s = number of likes */
+				__( '%s like', 'rest-likes' ),
+				$likes
+			);
```
Comment (ocean90) [inline inc/Controller.php:528]:
```suggestion
			$likes_text = __( 'One like', 'rest-likes' );
```

Die Überprüfung auf 1 erlaubt uns extra einen anderen Text ohne Platzhalter zu werden. Das sollte überall geändert werden.

File: inc/Controller.php [hunk 4]
```diff
@@ -520,13 +520,28 @@ public function get_like_count( $object_id ) {
...
+		} else {
+			$likes_text = sprintf(
+				/* translators: $s = number of likes */
```
Comment (ocean90) [inline inc/Controller.php:531]:
```suggestion
				/* translators: %s: Number of likes */
```

Das sollte ebenfalls überall geändert werden.

**Review by ocean90 — CHANGES_REQUESTED** [review]

File: inc/Controller.php (hunk 3 above)
Comment (krokodok) [reply inc/Controller.php]: Schau mal hier: https://github.com/wearerequired/rest-likes/pull/148/commits/6aa53e65aa4035482acfc95be38fc7b30c32e051

File: inc/Controller.php (hunk 4 above)
Comment (krokodok) [reply inc/Controller.php]: Siehe: https://github.com/wearerequired/rest-likes/pull/148/commits/6aa53e65aa4035482acfc95be38fc7b30c32e051

**Review by ocean90 — APPROVED** [review]

**Pull request merged**